scilpy/io/utils.py

```diff
@@ -247,36 +247,87 @@ def add_bbox_arg(parser):
                              'streamlines).')
 
 
-def add_sh_basis_args(parser, mandatory=False):
-    """Add spherical harmonics (SH) bases argument.
+def add_sh_basis_args(parser, mandatory=False, input_output=False):
+    """
+    Add spherical harmonics (SH) bases argument.
 
     Parameters
     ----------
     parser: argparse.ArgumentParser object
         Parser.
     mandatory: bool
         Whether this argument is mandatory.
+    input_output: bool
+        Whether this argument should expect both input and output bases or not.
+        If set, the sh_basis argument will expect first the input basis,
+        followed by the output basis.
     """
-    choices = ['descoteaux07', 'tournier07']
-    def_val = 'descoteaux07'
+    if input_output:
+        nargs = 2
+        def_val = ['descoteaux07_legacy', 'tournier07']
+        input_output_msg = '\nBoth the input and output bases are ' +\
+                           'required, in that order.'
+    else:
+        nargs = 1
+        def_val = ['descoteaux07_legacy']
+        input_output_msg = ''
+
+    choices = ['descoteaux07', 'tournier07', 'descoteaux07_legacy',
+               'tournier07_legacy']
     help_msg = 'Spherical harmonics basis used for the SH coefficients. ' +\
-               '\nMust be either \'descoteaux07\' or \'tournier07\'' +\
+               input_output_msg +\
+               '\nMust be either \'descoteaux07\', \'tournier07\', \n' +\
+               '\'descoteaux07_legacy\' or \'tournier07_legacy\'' +\
                ' [%(default)s]:\n' +\
-               '    \'descoteaux07\': SH basis from the Descoteaux et al.\n' +\
-               '                      MRM 2007 paper\n' +\
-               '    \'tournier07\'  : SH basis from the Tournier et al.\n' +\
-               '                      NeuroImage 2007 paper.'
+               '    \'descoteaux07\'       : SH basis from the Descoteaux et al.\n' +\
+               '                             MRM 2007 paper\n' +\
+               '    \'tournier07\'         : SH basis from the new Tournier et al.\n' +\
+               '                             NeuroImage 2019 paper, as in MRtrix 3.\n' +\
+               '    \'descoteaux07_legacy\': SH basis from the legacy Dipy\n' +\
+               '                             implementation of the Descoteaux et al.\n' +\
+               '                             MRM 2007 paper\n' +\
+               '    \'tournier07_legacy\'  : SH basis from the legacy Tournier et al.\n' +\
+               '                             NeuroImage 2007 paper.'
 
     if mandatory:
         arg_name = 'sh_basis'
     else:
         arg_name = '--sh_basis'
 
-    parser.add_argument(arg_name,
+    parser.add_argument(arg_name, nargs=nargs,
                         choices=choices, default=def_val,
                         help=help_msg)
 
 
+def interpret_sh_basis(args):
```

Comment:
`parse_sh_basis_arg` or something? I'm just wondering if it's the best name for this function. (maybe it is!)